src/preprocess.py

```diff
@@ -58,14 +100,21 @@ def remove_stop_words(collection: Dict[str, List[str]], stop_word_path: str) ->
     Returns:
         {Dict[str, List[str]]} -- updated corpus
     """
-    with open(stop_word_path,"r") as f:
-        stp = [word.lower() for word in f.read().split("\n") if word != ""]
+    stp = load_stop_words(stop_word_path)
     new_corpus = {}
     for key in collection.keys():
-        new_corpus[key] = [word for word in collection[key] if word not in stp]
+        new_corpus[key] = remove_stop_words_from_document(collection[key], stp, [])
     return new_corpus
 
-def collection_lemmatize(segmented_collection: Dict[str, List[str]]) -> Dict[str, List[str]]:
+def get_lemmatizer() -> StemmerI:
+    return WordNetLemmatizer()
+
+def lemmatize_one_token(token: str, lemmatizer: StemmerI) -> str:
```

Comment:
Pos tagging also use context to assign grammatical classe, it makes no sense to use it to lemmatize only one token. In the lemmatize collection function you can see how complete sentences are passed to the pos tagger before applying lemmatization.

Reply:
done